### ISSUE TITLE
bc: document in pod

### ISSUE THREAD

**Issue by jgpuckering** [briandfoy/PerlPowerTools#168]
The bc utility has documentation that is no more than the name and a description.  It needs to be filled in, preferably based on the GNU and POSIX documentation.

**Comment by jgpuckering** [comment]
The PPT version is mostly GNU and POSIX compliant, but not entirely.  That's why I spent the hours I did going through both docs to craft the one I added to the PPT bc POD.  I left in, but commented around, things that didn't work, like function and iscale and oscale.  That way if anyone has the wherewithal to fix any of those issues (I don't) then those bits of documentation can be restored.

In my view, it's better to have documentation that is accurate and relevant than to point someone off to related documentation that may mislead them or result in bug reports.  If you want bug reports for the things I documented around, I can certainly enter them.  But I personally am not going to invest the time to fix them -- not on this piece of code, which is messy and which likely no one even uses otherwise you'd have seen reports about functions being broken years ago.  (Recall that I tested the CPAN 0.13 version that was used as the basis for the PPT version and functions don't work in it either.)

I think if I were you I'd accept the POD code I wrote, since it's accurate and much better than the blank doc it replaces.  That way it at least gets into the commit history.  That goes for bc.t as well (vis a vis #167).

If you have solutions you prefer for either (or both), then just make them over top of my code.  Nothing I've done in these commits affects operational use of bc, so it's safe to release -- or you can wait releasing until you have your own solutions in place.   I clearly don't understand the rationale for your solutions well enough to replace mine with some rendering of what you counter-propose.  I suspect that will just result in more churn.  

To sum up, my concerns with your proposals are:

1. Inaccurate documentation will result in user frustration and possibly bug reports (re POD)
2. Executing bc on each loop of the test script will result in a 4 second run time, and I suspect you can't afford that kind of time when smoke testing 120+ scripts unless it's absolutely necessary.

Anyway, you are the code base owner so it's your call.  As I see it, your options are to reject my changes, or to accept them and then later re-fix them with what you have in mind.

**Comment by briandfoy** [comment]
This is just about the Pod. When we get everything else figured out, I'll merge the Pod that we both have. Right now let's get the other stuff merged and then come back to this.

**Comment by jgpuckering** [comment]
These changes (bc.t and bc's pod) were all done on different commits.  Are you not able to accept them separately?
I'm getting the sense that GitHub pull requests are all or nothing.  Either you accept all the commits related to the request, or none of them.  Is that correct?
Is it that you really *needed* (not just wanted) separate pull requests in order to be able to decide whether to take or reject them (independently) and to tie that to the closing of the related Issue(s).  Is that correct?
I did warn you that I'm a novice at GitHub collaboration workflow.  And now I'm a bit mystified as to how I should work on, say, two different things -- in code that doesn't even have dependencies -- without having to push separately, do a pull request for one of them, and then have to wait until that pull request was satisfied before doing a pull request on the second.  Obviously I need to minimize your work, but I'd like to be able to work on something else while waiting on a pull request to be accepted or rejected.  Otherwise what's the point of all this process machinery?
It seems to me I'd be better off creating separate branches in my repo for each Issue.  Then we just have to agree on the rules for scoping an issue.  For example, one issue for every missing or incomplete POD?  Or one to cover a whole bunch of them?  The latter seems to me to be perfectly workable given that POD mergers wouldn't likely conflict with other work -- especially in files that have virtually no POD.  In contrast, adding --help and -?/--usage to each script is less clear since it affects runtime and options processing is all over the place.
Anyway, I'm really not sure how to accomplish what you want me to do.  What steps do I need to follow in my fork of this code in order to provide what you need to move forward?  
    On Monday, June 19, 2023 at 01:35:31 p.m. EDT, brian d foy ***@***.***> wrote:  
 
 


This is just about the Pod. When we get everything else figured out, I'll merge the Pod that we both have. Right now let's get the other stuff merged and then come back to this.

**Comment by briandfoy** [comment]
If you want to talk about the pull request, let's do that in the pull request thread, not in this issue about Pod. It looks like you are replying to emails, which is a poor way to handle this. Go to the GitHub repo and reply in the relevant thread.

Don't worry about any of this. I'll fix it up next week.

**Comment by briandfoy** [comment]
This merges both versions of our docs. Thanks,
